### DISCUSSION TITLE
emitSchemaFile in federation

### DISCUSSION THREAD

**Discussion by JipSterk** [MichalLytek/type-graphql#877] [Q&A]
The output of `emitSchemaFile` is different then when i use an introspection tool to download the schema like [rover](https://github.com/apollographql/rover).

for instance i have an entity in the schema `User`

My model looks something like this:
```ts
@Directive(`@key(fields: "id")`)
@ObjectType()
export class User {
  @Field(() => ID)
  id: string;
}
```

in the schema generated by `type-graphql` it would look something like this:

```gql
directive @extends on INTERFACE | OBJECT

directive @external on FIELD_DEFINITION | OBJECT

directive @key(fields: String!) on INTERFACE | OBJECT

directive @provides(fields: String!) on FIELD_DEFINITION

directive @requires(fields: String!) on FIELD_DEFINITION

type User {
  id: ID!
}

type Query
```

the `directives` are from the `graphql` and `@apollo/federation` packages

but if i download the schema via introspection with `rover` with `rover subgraph introspect http://localhost:4000/graphql > schema.gql`

the schema would look like this:

```gql
type User @key(fields: "id") {
  id: ID!
}

type Query
```

if providing a repo would help shed some light on this i would happily provide it.

**Comment by MichalLytek** [comment]
Please read the docs carefully, it's written there why the emitted schema does not contain directives on types and fields.

I've prepared docs to not have to answer to such questions and use my time in a more productive manner 😉

**Reply by JipSterk** [reply]
ouch... im sorry i asked. next time ill read the docs more thorough 😅

**Reply by MichalLytek** [reply]
https://typegraphql.com/docs/directives.html

![image](https://user-images.githubusercontent.com/10618781/117141138-7d1e8900-adae-11eb-8c42-249fd29d0187.png)

**Reply by MichalLytek** [reply]
For directives and federation purposes you can `import { printSchema } from "@apollo/federation";`
https://github.com/MichalLytek/type-graphql/blob/master/examples/apollo-federation/helpers/buildFederatedSchema.ts
